arwen/host/arwen.go

```diff
@@ -372,6 +372,15 @@ func (host *vmHost) hasRetriableExecutionError(vmOutput *vmcommon.VMOutput) bool
 	return vmOutput.ReturnMessage == "allocation error"
 }
 
+// AreInSameShard returns true if the provided addresses are part of the same shard
+func (host *vmHost) AreInSameShard(leftAddress []byte, rightAddress []byte) bool {
```

Comment:
is this called always by sender SC and receiver SC ? hopefully not sender address and SC.

Reply:
There was a situation where `AreInSameShard()` was indeed called incorrectly, between the original TX caller and the receiver SC. Fixed.